### PR TITLE
Define a more formalized app structure

### PR DESCRIPTION
**Pull request by janthonysantana**
Based on this diagram: https://drive.google.com/file/d/12GHnIOIvkTfUev6mayjm51P6d7Z855Au/view?usp=sharing

Let me know if you have any feedback or questions!

### CODE REVIEW

File: components/category_list.py
Comment (yumidev) [inline components/category_list.py:1]: We might not need this component. Instead, we can make news_card component with category attribute and the app applies the filter to cards using the attribute and show only filtered cards in news_grid.

File: controllers/prompt_handler.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+   "account": os.getenv("SNOWFLAKE_ACCOUNT"),
+   "user": os.getenv("SNOWFLAKE_USER"),
+   "password": os.getenv("SNOWFLAKE_PASS"),
+   "warehouse": os.getenv("SNOWFLAKE_WH")
+ }
+
+session = Session.builder.configs(connection_parameters).create()  
+chosen_model = "snowflake-arctic"
+
+def get_cortex_response(summarize_text):
```
Comment (yumidev) [inline controllers/prompt_handler.py:20]:
```suggestion
summarize_request_str = "Summarize this text in one or two sentences:"
default_request = summarize_prompt_str

def get_cortex_response(summarize_text, request = default_request):
```

File: controllers/prompt_handler.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    if not summarize_text or not isinstance(summarize_text, str):
+        raise ValueError("Prompt Handler did not receive a valid text for summarization.")
+    
+    summarize_text = summarize_text.replace("'", "\\'")
+    prompt = f"Summarize this text in one or two sentences: {summarize_text}"
```
Comment (yumidev) [inline controllers/prompt_handler.py:25]:
```suggestion
    prompt = f"{request} {summarize_text}"
```

File: controllers/prompt_handler.py (hunk 1 above)
Comment (yumidev) [reply controllers/prompt_handler.py:20]: I suggest changing the `summarize_text` to `user_input`

File: controllers/prompt_handler.py
Comment (yumidev) [inline controllers/prompt_handler.py:1]: Alternatively, we can name this handler to `summarize_handler`

File: components/category_list.py
Comment (janthonysantana) [reply components/category_list.py:1]: Certainly, there's probably some files we don't need here, just an initial vision. We can get rid of it!

**Pull request merged**